### PR TITLE
feat(portals): add portals component

### PR DESCRIPTION
**Pull request by rdy**
[#95556026]

Signed-off-by: Ryan Dy rdy@pivotal.io

### CODE REVIEW

**Comment by rdy** [comment]
@stubbornella @gpleiss @vinsonchuong would someone mind reviewing this? also let us know if you'd like us to cross-team pair to build the style-guide component that demonstrates the use of portals?

**Comment by stubbornella** [comment]
I'll add a story for this.

File: components/portals/portals.js [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+function reset() {
+  emitter.removeAllListeners();
+  destinationPortals = {};
+}
+
+/**
+ * @component PortalSource
+ * @description HTML Rendered to PortalSource will be transported to PortalDestination
+ *
+ * @property name {String} Name for linking source and destination portals
```
Comment (gpleiss) [inline components/portals/portals.js]: Could we rename this property to `dest`? That way it seems more like the source is pointing to the dest.

**Comment by gpleiss** [comment]
This has been merged as 281a09329b41e031b97527999443b59d96f21e78

**Pull request closed without merging**